statusHandler/prometheusHandler_test.go

```diff
@@ -1,14 +1,30 @@
 package statusHandler_test
 
 import (
+	"flag"
+	"net/http"
+	"net/http/httptest"
 	"testing"
 
 	"github.com/ElrondNetwork/elrond-go/core"
 	"github.com/ElrondNetwork/elrond-go/statusHandler"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
 	prometheusUtils "github.com/prometheus/client_golang/prometheus/testutil"
 	"github.com/stretchr/testify/assert"
 )
 
+func init() {
+	// check if bench cli flag is set in order to init the prometheus server
+	flag.Parse()
+	bench := flag.CommandLine.Lookup("test.bench")
+	if bench.Value.String() != "" {
+		http.Handle("/metrics", promhttp.Handler())
+		go func() {
+			_ = http.ListenAndServe(":2112", nil)
```

Comment:
We should not use a real http server in unit tests or benchmarks. We can use the `httptest` package for it: https://golang.org/pkg/net/http/httptest/

Reply:
done, hopefully